libraries/botframework-connector/package.json

```diff
@@ -28,11 +28,13 @@
   },
   "dependencies": {
     "@azure/ms-rest-js": "1.9.1",
+    "@azure/msal-node": "^1.2.0",
     "@types/jsonwebtoken": "7.2.8",
     "@types/node": "^10.17.27",
     "adal-node": "0.2.2",
     "axios": "^0.21.1",
     "base64url": "^3.0.0",
+    "botbuilder-dialogs-adaptive-runtime-core": "4.1.6",
```

Comment:
I think adding a reference from connector to runtime is not ideal in terms of layer responsibilities. I understand you added this because the credentials factory receives the configuration. However, in C# this makes sense since configuration is part of the framework and does not force the connector -> runtime dependency. I'll look a bit more but sharing this early as an FYI.

Reply:
That's a fair criticism. Since we don't have as many built-ins in JS, those types live in this runtime core package. They could probably be migrated to the proper `botbuilder-core` package instead. I think that was more a function of utility during the runtime work rather than a fully baked design decision.

Reply:
Some alternatives could include moving the factory to the runtime or another layer, or receiving the app id instead of iconfiguration into the factory. I'll keep thinking of this but brainstorming here as I get more info.

Reply:
We wrote at the same time, thanks for the input. Seems like botbuilder-core is option 3 then.

Reply:
For more context (which you already know pretty sure), Connector is supposed to be a foundational layer and in general higher layers (bot builder) should depend on it. So having connector depending on the top layer is something we want to avoid, not only because of general purpose but because it could bring trouble down the line.

I don't have much problem with either of the options I and you laid out. Actually, your botbuilder-core proposal, if possible, seems among the most correct.

Reply:
Unfortunately, `botbuilder-core` already imports `botframework-connector` so moving types to `botbuilder-core` would introduce a cyclical reference. For now I've made the constructor accept the `appId` directly instead of the configuration.